umich_api/api_utils.py

```diff
@@ -25,7 +26,7 @@ class ApiUtil():
         Must init with the appropriate values
     """
 
-    def __init__(self, base_url, client_id, client_secret, api_json = None):
+    def __init__(self, base_url, client_id, client_secret, api_json = None, token_expires_percent=5):
```

Comment:
I like the idea of the percentage

Reply:
Yeah I think it worked out pretty good too. It's a percentage of time from the end, so 3600 seconds at 5% gives you 360 seconds from the end before it "expires"

So even if it has a 60 second token (unlikely) 5% would give you 3 seconds from the end. Probably not great to use 5% on a 60 second token but that's an application decision.